src/client/connect/http.rs

```diff
@@ -592,8 +592,8 @@ fn connect(
         .map_err(ConnectError::m("tcp set_nonblocking error"))?;
 
     if let Some(dur) = config.keep_alive_timeout {
-        if socket.set_keepalive(Some(dur)).is_err() {
-            warn!("tcp set_keepalive error");
+        if let Err(e) = socket.set_keepalive(Some(dur)) {
+            warn!("tcp set_keepalive error: {}", e);
         }
```

Comment:
Hey @seanmonstar. Thanks for the review. As I mentioned earlier, I am not sure if not returning an error when "set_keepalive" fails is the right thing to do. Isn't it might be critical to the client?

Reply:
Hm, I don't think it should error... do you have a more detailed argument/reason to support that it should?

Reply:
I thought for a second that it might be critial to keep keepalive low to probe the connection more frequently but I think you are right. Warning is enough.